vcs/gitcmd/repo.go

```diff
@@ -221,6 +235,30 @@ func (r *Repository) Branches(opt vcs.BranchesOptions) ([]*vcs.Branch, error) {
 	return branches, nil
 }
 
+func (r *Repository) branchesWithCommit(commitID string) (map[string]bool, error) {
+	cmd := exec.Command("git", "branch", "--contains", commitID)
+	cmd.Dir = r.Dir
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(string(out), "\n")
```

Comment:
Given that this operation might be called a lot, I think it's worth it to avoid the string conversion.

```go
lines := bytes.Split(out, []byte("\n"))
```

Also, can you preallocate the len of `branches`? When is `len(line) < 2` (if it just occurs once, then you can; otherwise fine to leave as-is).